.github/workflows/build_and_run.yml

```diff
@@ -0,0 +1,28 @@
+name: Build and Test
+
+on: [push, pull_request]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11"]
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
```

Comment:
```suggestion
        uses: actions/setup-python@v5
```

Reply:
you can also cache your dependencies by adding this to the setup-python action:

```yml
    cache: 'pip'
    cache-dependency-path: '**/requirements.txt'
```

More details here: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages